### DISCUSSION TITLE
Is there a callback after calling `set()`?

### DISCUSSION THREAD

**Discussion by zerosrat** [pmndrs/zustand#811] [Q&A] [answered]
same to react setState API: `setState(updater, [callback])`

**Comment by dai-shi** [comment, marked as answer]
In terms of store, `set()` is sync, so you can call any functions after that. What's your use case?

**Reply by zerosrat** [reply]
I'm using zustand to manage the state of my app. When an user launch the app, a full screen loading component will be displayed before  getting and setting data. I want to make sure data will be rendered on the screen after data is set. In react app, hiding loading component in the callback fn of `setState` API is the general way to do to this.

So I wonder Is there a callback after calling `set()`in zustand? If `set()` is sync, can I just hide loading after calling `set()`?

**Reply by dai-shi** [reply]
There's a little delay from setting the state, and visible on screen. If you want make sure if it's after showing the screen, you'd need to use `useEffect`. But, I suppose it's rare and your use case sounds just fine to assume when `set()` is done, it will be shown on the screen in the next render. So, yes, you can just hide loading after calling `set()`.

**Reply by zerosrat** [reply]
Thanks a lot.

**Reply by dw-herrmann** [reply]
@ zerosrat, maybe its a little late, but there is a ["subscribe"-feature](https://github.com/pmndrs/zustand#readingwriting-state-and-reacting-to-changes-outside-of-components).
